src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -141,20 +142,20 @@ def _handle_command_package_error(self, err_msg):  # pylint: disable=no-self-use
                 pass
 
     def validation_error(self, message):
-        from azure.cli.core.telemetry import log_telemetry
-        log_telemetry('validation error', log_type='trace', prog=self.prog)
+        telemetry.set_user_fault('validation error')
         return super(AzCliCommandParser, self).error(message)
 
     def error(self, message):
-        from azure.cli.core.telemetry import log_telemetry
-        log_telemetry('parse error', message=message, prog=self.prog)
+        telemetry.set_user_fault('parse error: {}'.format(message))
         self._handle_command_package_error(message)
         return super(AzCliCommandParser, self).error(message)
 
     def format_help(self):
-        from azure.cli.core.telemetry import log_telemetry
         is_group = self.is_group()
-        log_telemetry('show help', prog=self.prog)
+
+        telemetry.set_command_details(command=self.prog[3:])
```

Comment:
What's the significance of the 3 here?
Before, we were including the full `self.prog`.